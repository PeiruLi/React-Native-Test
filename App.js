import React, {Component} from 'react';
import {StyleSheet, Text, View, TextInput, Button} from 'react-native';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
    flex: 1,
    justifyContent: 'center',
  },
  instructions: {
    color: '#333333',
    marginBottom: 5,
    textAlign: 'center',
  },
  welcome: {
    fontSize: 20,
    margin: 10,
    textAlign: 'center',
  },
  Button: {
    color: "#841584"
  }
});

export default class App extends Component {

  constructor(props) {  
    super(props);  
    // 初始状态  
    this.state = { 
      username: 'Username',
      password: 'Password',
      label: 'Place',
      title: 'Click Me'
     };
     this.onPressLearnMore.bind(this);
    } 

  onPressLearnMore(e) {
    console.log();
    this.setState({
      title: 'Learn More'
    });
  }

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>Welcome to React Native!</Text>
        <Text style={styles.instructions}>
          This is a React Native snapshot test.
        </Text>

        <View style={{flexDirection:'row'}}>
            <Text style={{flex:0.2,alignSelf:'center',fontSize:16}}>
              Username
            </Text>
            <TextInput
              style={{height: 40, flex: 0.6, borderColor: 'gray', borderWidth: 1}}
              onChangeText={(text) => this.setState({text})}
              placeholder={this.state.username}
            />
        </View>

        <View style={{flexDirection:'row'}}>
            <Text style={{flex:0.2,alignSelf:'center',fontSize:16}}>
              Password
            </Text>
            <TextInput
              style={{height: 40, flex: 0.6, borderColor: 'gray', borderWidth: 1}}
              onChangeText={(text) => this.setState({text})}
              placeholder={this.state.password}
            />
        </View>
        <Button
            style={styles.Button}
            onPress={(event) => this.onPressLearnMore(event, this.props.title)}
            title= {this.state.title}
            accessibilityLabel="Tap Me"
          />
        
      </View>
    );
  }
}