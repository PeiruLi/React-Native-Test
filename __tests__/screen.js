import 'react-native';
import React from 'react';
import App from '../App';
import { shallow } from 'enzyme';

import renderer from 'react-test-renderer';

describe('root', function () {
  test('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    // console.log(tree.children);
    expect(tree).toMatchSnapshot();
  });

  test('renders as expected', () => {
    const wrapper = shallow(<App />);
    console.log(wrapper.find('Text').get(0).props.children);
    expect(wrapper.find('Text').get(0).props.children).toBe('Welcome to React Native!');
  });

  test('Click Button', () => {
    const wrapper = shallow(<App />);
    wrapper.find('Button').simulate('press');
    console.log(wrapper.find('Button').get(0).props.title);
    expect(wrapper.find('Button').get(0).props.title).toBe('Learn More');
  });

});