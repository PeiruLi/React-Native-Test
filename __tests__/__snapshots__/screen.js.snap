// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to React Native!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    This is a React Native snapshot test.
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "flex": 0.2,
          "fontSize": 16,
        }
      }
    >
      Username
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Username"
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "flex": 0.6,
          "height": 40,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "flex": 0.2,
          "fontSize": 16,
        }
      }
    >
      Password
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Password"
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "flex": 0.6,
          "height": 40,
        }
      }
    />
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel="Tap Me"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Click Me
      </Text>
    </View>
  </View>
</View>
`;

exports[`root renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to React Native!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    This is a React Native snapshot test.
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "flex": 0.2,
          "fontSize": 16,
        }
      }
    >
      Username
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Username"
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "flex": 0.6,
          "height": 40,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "flex": 0.2,
          "fontSize": 16,
        }
      }
    >
      Password
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Password"
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "flex": 0.6,
          "height": 40,
        }
      }
    />
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel="Tap Me"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Click Me
      </Text>
    </View>
  </View>
</View>
`;
